From 914b58be8d2826b61ff0f154aecf8bc559803cb7 Mon Sep 17 00:00:00 2001
From: Stella Laurenzo <stellaraccident@gmail.com>
Date: Fri, 21 Feb 2025 13:17:39 -0800
Subject: [PATCH 2/3] Use proper casing for 'hip' and 'hiprtc' packages.

* Package names are case-sensitive, and the name for both of these is lowercase.
* There are situations where the wrong casing can be made to work (i.e. if you manually specify "-DHIP_DIR=" and "-DHIPRTC_DIR=" config vars with the wrong case, the search procedure will happen to work in this very narrow case, but it isn't right).
* For legacy reasons, ROCM actually does *also* have a "HIP" (uppercase) package (a "module" package, not a "config" package, which is what is saving the day here because the "config" version is being requested). The right one for this case is "hip".

References:

* Docs: https://rocm.docs.amd.com/en/latest/conceptual/cmake-packages.html#consuming-the-hip-api-in-c-code
* CMake find_package search procedure for prefix searches in config mode: https://cmake.org/cmake/help/latest/command/find_package.html#config-mode-search-procedure
* Concrete locations of config files in the SDK: `lib/cmake/hiprtc/hiprtc-config.cmake` and `lib/cmake/hip/hip-config.cmake`
---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 53873b78..9559e515 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -193,8 +193,8 @@ rocm_check_target_ids(AMDGPU_TARGETS TARGETS "${AMDGPU_TARGETS}")
 set(GPU_TARGETS "${AMDGPU_TARGETS}" CACHE STRING "GPU architectures to build for")
   
 # HIP is required - library and clients use HIP to access the device
-find_package( HIP REQUIRED CONFIG )
-find_package( HIPRTC REQUIRED CONFIG )
+find_package( hip REQUIRED CONFIG )
+find_package( hiprtc REQUIRED CONFIG )
 
 # The nvidia backend can be used to compile for CUDA devices.
 # Specify the CUDA prefix in the CUDA_PREFIX variable.
-- 
2.47.1.windows.2

